extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/utils/GeometryUtils.java

```diff
@@ -0,0 +1,236 @@
+package mil.nga.giat.geowave.adapter.vector.utils;
+
+import java.awt.geom.Point2D;
+import java.util.Arrays;
+import java.util.Collections;
+
+import javax.measure.unit.SI;
+import javax.measure.unit.Unit;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.log4j.Logger;
+import org.geotools.ows.bindings.UnitBinding;
+import org.geotools.referencing.GeodeticCalculator;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.referencing.cs.CoordinateSystem;
+import org.opengis.referencing.operation.TransformException;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Envelope;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.LineString;
+import com.vividsolutions.jts.geom.Point;
+
+public class GeometryUtils
+{
+
+	private static Logger LOGGER = Logger.getLogger(GeometryUtils.class);
+
+	/**
+	 * Build a buffer around a geometry for the pro
+	 * 
+	 * @param crs
+	 * @param geometry
+	 * @param distanceUnits
+	 * @param distance
+	 * @return
+	 * @throws TransformException
+	 */
+	public static final Pair<Geometry, Double> buffer(
+			final CoordinateReferenceSystem crs,
+			final Geometry geometry,
+			final String distanceUnits,
+			final double distance )
+			throws TransformException {
+		Unit<?> unit;
+		try {
+			unit = (Unit<?>) new UnitBinding().parse(
+					null,
+					distanceUnits);
+		}
+		catch (final Exception e) {
+			unit = SI.METER;
+			LOGGER.warn(
+					"Cannot lookup unit of measure " + distanceUnits,
+					e);
+		}
+		final double meterDistance = unit.getConverterTo(
+				SI.METER).convert(
+				distance);
+		final double degrees = distanceToDegrees(
+				crs,
+				geometry,
+				meterDistance);
+		// buffer does not respect the CRS; it uses simple cartesian math.
+		return Pair.of(
+				adjustGeo(
```

Comment:
I'd forgo this call to adjustGeo()

I don't think it quite does what we want it to do - the way I read it, the geometry will self-correct for dateline crossing by converting values outside of the range of valid longitude values to be the reflection of that longitude value (ie. converting -181 to 179 for example).

The problem is that JTS is working in 2D cartesian space and this won't really result in a geometry that represents  what you want it to.  In fact I think it may often result in exceptions from JTS because the geometry will be invalid (folding across itself).  What is really need to correctly represent the geometry is to cut it into multiple geometries, on each side of the dateline (in some cases more than one in fact, if you think of a zigzag, 'Z' shape as an example).  I've written some code to basically do that, I think its out of scope for this right now though and I'd just get rid of adjustGeo().